modules/cognito_userpool/main.tf

```diff
@@ -0,0 +1,40 @@
+resource "aws_cognito_user_pool" "pool" {
+  name = "DataSimulationPool"
+  password_policy {
+    minimum_length    = 8
+    require_lowercase = true
+    require_uppercase = false  
+    require_symbols   = false
+    require_numbers   = true
+  }
+
+  device_configuration {
+    challenge_required_on_new_device = false
+  }
+
+  username_configuration {
+    case_sensitive = false
+  }
+
+  admin_create_user_config {
+    allow_admin_create_user_only = true
+  }
+}
+
+resource "aws_cognito_user_pool_client" "client" {
+  name            = "SimulationApp"
+  user_pool_id    = aws_cognito_user_pool.pool.id
+  generate_secret = false
+}
+
```

Comment:
We also need the cognito url so that requires a domain like so?

```tf
resource "aws_cognito_user_pool_domain" "api_pool_domain" {
  domain = aws_cognito_user_pool_client.api_pool_client.id
  user_pool_id = aws_cognito_user_pool.api_pool.id
}
```

Reply:
done